recipes/automake/all/conanfile.py

```diff
@@ -11,7 +11,7 @@ class AutomakeConan(ConanFile):
     exports_sources = ["patches/**"]
     license = ("GPL-2.0-or-later", "GPL-3.0-or-later")
 
-    settings = "os"
+    settings = "os", "arch", "compiler"
```

Comment:
Why not adding `build_type` as well? I think the take away was that adding all settings is much easier or probably more correct in many cases 🤔

Reply:
`automake` is a script-only package.
It is a collection of perl/m4/bash scripts.

Reply:
I see, still, I think that always adding all settings and removing them accordingly from package_id might simplify things.

Reply:
(Personal opinion) I think the same, I like to list everything in `settings` and then remove whatever is needed... and then Conan could add all the settings by default (if you already adds all of them to all the recipes... 🤷 ). And, if the user has custom settings with more first-level settings Conan will add them too and require a value for them in the input profiles (not sure if this is good or bad).

This is just a random thought, I haven't thought deeply about other side-effects.